In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Loading multiple CSV files
df1 = pd.read_csv('CAvideos.csv')
df2 = pd.read_csv('DEvideos.csv')
df3 = pd.read_csv('FRvideos.csv')
df4 = pd.read_csv('GBvideos.csv')
df5 = pd.read_csv('INvideos.csv')
df6 = pd.read_csv('JPvideos.csv', encoding='ISO-8859-1')
df7 = pd.read_csv('KRvideos.csv', encoding='ISO-8859-1')

df8 = pd.read_csv('MXvideos.csv', encoding='ISO-8859-1')

df9 = pd.read_csv('RUvideos.csv', encoding='ISO-8859-1')

df10 = pd.read_csv('USvideos.csv')

# Combine them into one DataFrame
df = pd.concat([df1, df2, df3, df4, df5, df6, df7, df8, df9, df10], ignore_index=True)

# Preview
df.head()


video_id trending_date  \
0  n1WpP7iowLc      17.14.11   
1  0dBIkQ4Mz1M      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  d380meD0W0M      17.14.11   
4  2Vv-BfVoq4g      17.14.11   

                                               title channel_title  \
0         Eminem - Walk On Water (Audio) ft. Beyoncé    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id              publish_time  \
0           10  2017-11-10T17:00:03.000Z   
1           23  2017-11-13T17:00:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-12T18:01:41.000Z   
4           10  2017-11-09T11:04:14.000Z   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncé i...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  I know it's been a while since we did this sho...  
4  🎧: https://ad.gt/yt-perfect\n💰: https://atlant...

* Step 1: Load and Combine YouTube Trending Datasets from Multiple Countries
In this step:

We imported essential libraries: pandas, numpy, matplotlib, and seaborn.

We loaded 10 CSV files representing YouTube trending videos from different countries like CA, DE, FR, GB, IN, JP, KR, MX, RU, and US.

Some files were encoded in ISO-8859-1 to handle special characters (e.g., Japanese, Russian).

Finally, we combined all datasets into a single DataFrame using pd.concat() to prepare for unified analysis.

We also displayed the first few rows using .head() to verify successful loading.

In [6]:
# Basic info
print(df.info())

# Check for missing values
print("\nMissing values:\n", df.isnull().sum())

# Check for duplicates
print("\nNumber of duplicate rows:", df.duplicated().sum())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375942 entries, 0 to 375941
Data columns (total 16 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   video_id                375942 non-null  object
 1   trending_date           375942 non-null  object
 2   title                   375942 non-null  object
 3   channel_title           375942 non-null  object
 4   category_id             375942 non-null  int64 
 5   publish_time            375942 non-null  object
 6   tags                    375942 non-null  object
 7   views                   375942 non-null  int64 
 8   likes                   375942 non-null  int64 
 9   dislikes                375942 non-null  int64 
 10  comment_count           375942 non-null  int64 
 11  thumbnail_link          375942 non-null  object
 12  comments_disabled       375942 non-null  bool  
 13  ratings_disabled        375942 non-null  bool  
 14  video_error_or_removed  375942 non-n

* Step 2: Data Exploration – Missing Values & Duplicates

The dataset has a total of 375,942 rows and 16 columns after combining all countries' video data.
We used .info() to check data types and found a mix of object, int64, and bool types.
Using .isnull().sum(), we discovered that only the description column has missing values (19,478 nulls), which is acceptable for text fields and can be optionally filled or dropped.
.duplicated().sum() showed 34,053 duplicate rows, which we can remove to ensure data integrity and avoid bias in our analysis.

In [7]:
# Drop duplicate rows
df.drop_duplicates(inplace=True)

# Optionally fill missing descriptions with an empty string
df['description'].fillna('', inplace=True)

# Confirm cleanup
print("Duplicates after cleaning:", df.duplicated().sum())
print("Missing values after cleaning:\n", df.isnull().sum())


Duplicates after cleaning: 0
Missing values after cleaning:
 video_id                  0
trending_date             0
title                     0
channel_title             0
category_id               0
publish_time              0
tags                      0
views                     0
likes                     0
dislikes                  0
comment_count             0
thumbnail_link            0
comments_disabled         0
ratings_disabled          0
video_error_or_removed    0
description               0
dtype: int64


* Data Cleaning Summary
Duplicates Removed: Successfully removed 34,053 duplicate rows from the combined dataset.

Missing Values Handled: The only missing data was in the description column (19,478 entries), which has now been filled (likely using a placeholder or imputation).

Now, there are no missing values in any column.

The dataset is now clean and ready for analysis or visualization.

In [8]:
df.to_csv('cleaned_youtube_data.csv', index=False)
